### PR TITLE
Some devops code and a MIT Licence

### PR DESCRIPTION
**Pull request by kern3020**
While on the sidelines of yesterday’s class I was able to talk with some people which has given my some ideas about what the computational environment should look like. Eric A. mentioned some arrangement with Oregon State. Dan said he had access to an HPC system. Our presenters mentioned Basespace. I will probably just use the google compute engine. Docker containers for the various assemblers (and other bioinformatics tools) will be very handy. 

Here you'll find. 
1) an MIT license. 
2) a docker file for the abyss assembler.
3) a shell script for launching a node via google compute engine.

### CODE REVIEW

File: cloud/gcp/gcp.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+HOST=mar
+# IMAGE="ubuntu-1604-xenial-v20171107b"
+IMAGE_FAMILY='ubuntu-1604-lts'
+IMAGE_PROJECT='ubuntu-os-cloud' # image-project 
+TYPE="n1-standard-1"
+ZONE="us-west1-a"
+
+if [ "$#" -lt 1 ] 
+then
+  echo "Usage: gce.sh {add-disk | ssh | start | stop}"
```
Comment (ArturKlauser) [inline cloud/gcp/gcp.sh]: Here it's "gce.sh" whereas the filename is gcp.sh.
(same below)

File: cloud/gcp/gcp.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  exit 1 
+fi
+
+CMD="$1"
+
+if [ ${CMD}x = "add-disk"x ]
```
Comment (ArturKlauser) [inline cloud/gcp/gcp.sh]: Suggesting using the more modern variant of conditions:
if [[ ${CMD} = "add-disk" ]]
(same below)

File: cloud/gcp/gcp.sh (hunk 1 above)
Comment (ArturKlauser) [inline cloud/gcp/gcp.sh]: "reset" handled below but not mentioned here.

File: cloud/gcp/gcp.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    gcloud compute disks create ${DISK_NAME} --size ${DISK_SIZE}GB --zone ${ZONE} 
+    ;; 
+  "reset")
+    gcloud compute instances reset ${HOST} --zone ${ZONE}
+    ;;
+  "ssh")   
+    gcloud compute ssh ${HOST} --zone ${ZONE} 
+    ;;
+  "start")
+    if [ ${CMD}x = "start"x ]
```
Comment (ArturKlauser) [inline cloud/gcp/gcp.sh]: You're already in the case where $CMD == "start", so this "if" is redundant.

File: cloud/gcp/gcp.sh [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+		 --disk "name=${2}" \
+		 --image-family ${IMAGE_FAMILY}  --image-project ${IMAGE_PROJECT}
+      fi
+    fi 
+
+    ;;
+  "stop") 
+    gcloud compute instances delete ${HOST} --zone ${ZONE} # --keep-disks boot
+    ;;
+  *) echo "unknown command, ${CMD}. Ignoring."
```
Comment (ArturKlauser) [inline cloud/gcp/gcp.sh]: Probably would want to "exit 1" from this.

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,9 @@
 /_book
 .DS_Store
 /node_modules
+
+# C extensions
+*.so
+
+# emacs backup
+*~
```
Comment (ArturKlauser) [inline .gitignore]: /n on last line too please.

**Comment by kern3020** [comment]
@ArturKlauser Thanks for the thoughtful comments!

**Comment by kern3020** [comment]
Last commit fixes issues brought up by the code review.

**Comment by ArturKlauser** [comment]
LGTM. @bstaats any more comments?

**Comment by kern3020** [comment]
@ArturKlauser and @bstaats Hello Gentlemen, So, what do you think? nag. nag. :-)

**Comment by ArturKlauser** [comment]
Sorry about the delay. I wanted to give Brian a chance to look it over, but he might be busy.

**Pull request merged**